airflow/models.py

```diff
@@ -1746,7 +1795,36 @@ def dry_run(self):
         task_copy.dry_run()
 
     @provide_session
-    def handle_failure(self, error, test_mode=False, context=None, session=None):
+    def handle_deploy_orphan(self, session=None):
+        self.log.warn("Marking task {} with state QUEUED because it was orphaned by a deploy.".format(self.task_id))
+        task = self.task
+        self.end_date = datetime.utcnow()
+        self.hostname = None
+        self.set_duration()
+        Stats.incr(
+            'deploy_zombies',
+            tags=[
+                'task_id:%s' % self.task_id,
+                'dag_id:%s' % self.dag_id,
+                'operator:%s' % task.__class__.__name__
+            ],
+        )
+
+        session.add(Log(State.NONE, self))
+
+        self.state = State.NONE
+
+        session.merge(self)
+        session.commit()
+
+    def handle_failure(self, error, test_mode=False, context=None):
+        self.handle_retry_event(error, test_mode, context, State.UP_FOR_RETRY)
```

Comment:
This preserves the old behavior everywhere this method was already being called.